### PR TITLE
Refactor module MaryTTS 

### PR DESCRIPTION
**Pull request by wardner**
Refactor @fonos/marytts to improve type safety and include unit testing
PF-84

### CODE REVIEW

**Comment by rihernandez** [comment]
Verified!

File: mods/marytts/src/mary_tts.ts [hunk 1]
```diff
@@ -1,28 +1,47 @@
...
 import http from "http";
 import fs from "fs";
 import path from "path";
 import {AbstractTTS, optionsToQueryString, computeFilename} from "@fonos/tts";
-import logger from "@fonos/logger";
+import {maryTtsLogs} from "./logs";
+import {DefaultConfig, OptionsInterface} from "./types";
 
 /**
- * @classdesc The default TTS engine in a Fonos deployment.
```
Comment (psanders) [inline mods/marytts/src/mary_tts.ts]: Revert to include the dot.

File: mods/marytts/src/mary_tts.ts [hunk 2]
```diff
@@ -31,59 +50,59 @@ export default class MaryTTS extends AbstractTTS {
    * Constructs a new MaryTTS object.
    *
    * @see module:tts:AbstractTTS
+   * @param {DefaultConfig} config - Configuration of the marytts
    */
-  constructor(options?: any) {
+  constructor(config?: DefaultConfig) {
```
Comment (psanders) [inline mods/marytts/src/mary_tts.ts]: Refactor `DefaultConfig` interface to have name `MaryTTSConfig`

File: mods/marytts/src/mary_tts.ts [hunk 3]
```diff
@@ -31,59 +50,59 @@ export default class MaryTTS extends AbstractTTS {
...
-    if (!opts.port) throw new Error("port field is required");
-
-    const q = `INPUT_TYPE=TEXT&AUDIO=WAVE_FILE&OUTPUT_TYPE=AUDIO&LOCALE=${opts.locale}`;
-    this.serviceUrl = `http://${opts.host}:${opts.port}/process?${q}`;
+  /**
+   * Init of the marytts constructor
+   *
+   * @param {DefaultConfig} config - Configuration of the marytts
+   */
+  init(config?: DefaultConfig): void {
```
Comment (psanders) [inline mods/marytts/src/mary_tts.ts]: Remove hardcode and make config mandatory.

File: mods/marytts/src/mary_tts.ts [hunk 4]
```diff
@@ -31,59 +50,59 @@ export default class MaryTTS extends AbstractTTS {
...
    * @inherit
+   *
+   * @param {string} text - Text that will be synthesized
+   * @param {OptionsInterface} options - Options of the marytts, locale and voice
+   * @return {Promise<String>}
+   * For more information: http://marytts.phonetik.uni-muenchen.de:59125/documentation.html
+   * Advice: On windows the command "which" that sox library uses is not the same. In windows is "where" instead
    */
-  synthesize(text: string, options: any = {}): Promise<string> {
+  synthesize(text: string, options?: OptionsInterface): Promise<string> {
```
Comment (psanders) [inline mods/marytts/src/mary_tts.ts]: Refactor `OptionsInterface` to have name `MarySynthOptions`

File: mods/marytts/test/tts.unit.test.ts [hunk 5]
```diff
@@ -10,52 +28,45 @@ import fs from "fs";
...
+  const defConfig = {
+    host: "localhost",
+    port: 59125,
+    locale: "EN_US"
+  };
+
+  sandbox.stub(MaryTTS.prototype, "init").returns();
 
-  it("rejects if the TTS engine response is not 200", () => {
+  it("Should rejects if the TTS engine response is not 200", async () => {
```
Comment (psanders) [inline mods/marytts/test/tts.unit.test.ts]: Start text it lowercase.

File: mods/marytts/test/tts.unit.test.ts [hunk 6]
```diff
@@ -10,52 +28,45 @@ import fs from "fs";
...
 
-    expect(tts.synthesize("hello world")).to.eventually.rejectedWith(
-      "Request failed status code"
-    );
+    const tts = new MaryTTS(defConfig);
 
+    await expect(tts.synthesize("hello world", {locale: "en_US"}))
+      .to.be.eventually.rejected.and.to.be.an.instanceOf(Error)
+      .to.have.property("message", "Request failed status code: 201");
     expect(pipe).to.not.have.been.called;
```
Comment (psanders) [inline mods/marytts/test/tts.unit.test.ts]: Be specific on how many times should they be called `pipe` and `createWriteStream`

File: mods/marytts/test/tts.unit.test.ts [hunk 7]
```diff
@@ -10,52 +28,45 @@ import fs from "fs";
...
-    };
-    get.yields(response);
-
-    const tts = new MaryTTS();
+    const get = sandbox.stub(http, "get").yields({statusCode: 200, pipe});
 
-    expect(tts.synthesize("hello world")).to.eventually.contain("/tmp/");
+    const tts = new MaryTTS(defConfig);
+    const synth = await tts.synthesize("hello world");
+    expect(synth).to.contain("tmp");
```
Comment (psanders) [inline mods/marytts/test/tts.unit.test.ts]: Refactored to include `/tmp/` instead of `tmp`

File: mods/dispatcher/src/dispatcher.ts [hunk 8]
```diff
@@ -18,11 +18,16 @@ if (process.env.NODE_ENV === "dev") {
 
 function dispatch(channel: any) {
   try {
+    const maryTTSConfig = {
```
Comment (psanders) [inline mods/dispatcher/src/dispatcher.ts:21]: In this case, you will have to use variables from the environment instead of hardcoded values.

**Pull request merged**